wsit/hudson/jenkins.sh

```diff
@@ -0,0 +1,361 @@
+
+
+cd $METRO_SVN_ROOT/wsit/hudson/
+source common.sh
+if [ -z "$M2_LOCAL_REPO" ]; then
+    export M2_LOCAL_REPO=$WORK_DIR/.repository
+fi
+
+#PROXY
+set_proxy
+
+export JAVA_MEMORY_PROP="-Xms256m -Xmx768m -XX:PermSize=256m -XX:MaxPermSize=512m"
+export ANT_OPTS="$JAVA_MEMORY_PROP $JAVA_PROXY_PROP"
+export MAVEN_OPTS="-Dmaven.repo.local=$M2_LOCAL_REPO -Dmaven.javadoc.skip=true $JAVA_MEMORY_PROP $JAVA_PROXY_PROP"
+
+source setenv.sh
+
+set_common
+print_env
+
+#prepare git source
+pushd $METRO_SVN_ROOT
+git reset --hard
+git clean -fd
+popd
+
+
+if [! -d "$GF_SVN_ROOT"  ]; then
+  git init $GF_SVN_ROOT
+  git fetch --tags --progress git@github.com:javaee/glassfish.git +refs/heads/*:refs/remotes/origin/*
+fi
+
+pushd $GF_SVN_ROOT
+git reset --hard
+git clean -fd
+popd
+
+
+#fallback to defaults if needed
+if [ -z "$MVN_REPO_URL" ]; then
+    MVN_REPO_URL="https://maven.java.net/content/groups/staging"
+fi
+
+if [ -z "$METRO_SVN_REPO" ]; then
+    METRO_SVN_REPO=$METRO_FORKED_REPO
+fi
+
+pushd $WORK_DIR
+
+#CLEAN UP
+rm -rf $GF_WORK_DIR
+rm -rf $METRO_WORK_DIR
+rm -f *.patch
+rm -f *.zip
+
+if [ ! -f "$GF_ZIP" -a ! -z "$GF_URL" ]; then
+  wget -N $GF_URL
+  export GF_ZIP=$WORK_DIR/${GF_URL##*/}
+fi
+
+echo "METRO_VERSION: $METRO_VERSION"
+echo "METRO_REVISION: $METRO_REVISION"
+echo "CTS_ZIP: $CTS_ZIP"
+
+if [[ ! -z $METRO_REVISION ]]; then #generate a temp metro version
+  DATESTAMP=`date +%y%m%d.%H%M`
+  BUILD_NUMBER=b${DATESTAMP}
+  METRO_VERSION=$METRO_MAJOR_VERSION-$BUILD_NUMBER
+fi
+
+METRO_BUNDLE="org/glassfish/metro/metro-standalone"
+if [ -z "$METRO_VERSION" ]; then
+    LATEST_METRO_BUILD=`curl -s -k $MVN_REPO_URL/$METRO_BUNDLE/maven-metadata.xml | grep "<version>$METRO_MAJOR_VERSION-b[0-9]" | cut -d ">" -f2,2 | cut -d "<" -f1,1 | tail -1 | cut -d "b" -f2,2`
+    METRO_VERSION="$METRO_MAJOR_VERSION-b$LATEST_METRO_BUILD"
+    echo "Latest metro build: " $LATEST_METRO_BUILD
+    echo "Metro version: " $METRO_VERSION
+fi
+
+if [ -z "$METRO_REVISION" ]; then
+    METRO_URL="$MVN_REPO_URL/$METRO_BUNDLE/$METRO_VERSION/metro-standalone-$METRO_VERSION.zip"
+    _wget $METRO_URL
+    export METRO_ZIP=$WORK_DIR/${METRO_URL##*/}
+fi
+
+if [ ! -f "$CTS_ZIP" ]; then
+    wget $CTS_URL
+    export CTS_ZIP=$WORK_DIR/${CTS_URL##*/}
+fi
+popd
+
+echo "GF_ZIP: $GF_ZIP"
+echo "METRO_ZIP: $METRO_ZIP"
+echo "CTS_ZIP: $CTS_ZIP"
+echo "Metro SVN root: $METRO_SVN_ROOT"
+
+echo "Preparing workspace:"
+#delete old
+pushd $WORK_DIR
+for dir in "test_results" ".repository" #"wsit"
+do
+    if [ -e "$dir" ] ; then
+        echo "Removing $dir"
+        rm -rf $dir
+    fi
+done
+
+popd
+
+
+cd $WORK_DIR
+if [ -z "$METRO_URL" ]; then
+    if [ ! -z "$METRO_REVISION" ]; then
+        rm -rf $METRO_SVN_ROOT || true
+        echo "Checking out Metro sources using revision: $METRO_REVISION"
+        git clone $METRO_SVN_REPO $METRO_SVN_ROOT
+        pushd $METRO_SVN_ROOT/wsit/
+        git checkout $METRO_REVISION
+        git pull
```

Comment:
no, remove the call to `git pull`. What is wanted here is to check given `$METRO_REVISION` and not something on top of it